core/flyout_vertical.js

```diff
@@ -287,11 +287,14 @@ Blockly.VerticalFlyout.prototype.isDragTowardWorkspace = function(
 };
 
 /**
- * Return the deletion rectangle for this flyout in viewport coordinates.
- * @return {?Blockly.utils.Rect} Rectangle in which to delete.
+ * Returns the bounding rectangle of the drag target area in pixel units
+ * relative to viewport.
+ * @return {Blockly.utils.Rect} The component's bounding box.
  */
 Blockly.VerticalFlyout.prototype.getClientRect = function() {
-  if (!this.svgGroup_) {
+  if (!this.svgGroup_ || this.autoClose || this.isVisible()) {
```

Comment:
A rect with dimensions zero would not work as cleanly because then they would not be able to be filtered out at `recordDragTargets` and, in theory, something could be able to be over them when they should instead be ignored.